retrace/glstate_shaders.cpp

```diff
@@ -419,6 +419,14 @@ dumpUniformBlock(StateWriter &writer,
     if (raw_data) {
         std::string qualifiedName = resolveUniformName(name, size);
 
+        GLuint actual_count = mapping.getSize() / array_stride;
```

Comment:
Thanks for looking into this.  I agree extra checking is a good thing.

But I fear this check might be a tad too strict, as I'm not sure that the array elements are guaranteed to be contiguous in memory.

Imagine a buffer with 12 bytes, with elements of 4 bytes, and a stride of 8 bytes.  There's space for two elements (with 4 bytes of padding.) But merely diving buffer size by stride would say only one element can fit...

Perhaps this can never happen -- it's hard to say if that's legal or not --, but the safest thing to do would be to pass the buffer size down `dumpAttribArray` and `dumpAttrib`, and do the checking there.

Another alternative would be to try to compensate for this when computing `actual_count`.